zk-token-sdk/src/encryption/discrete_log.rs

```diff
@@ -107,34 +107,37 @@ impl DiscreteLog {
         Self {
             generator,
             target,
-            num_threads: 1,
-            range_bound: TWO16 as usize,
+            num_threads: None,
+            range_bound: (TWO16 as usize).try_into().unwrap(),
             step_point: G,
-            compression_batch_size: 32,
+            compression_batch_size: 32.try_into().unwrap(),
         }
     }
 
     /// Adjusts number of threads in a discrete log instance.
     #[cfg(not(target_arch = "wasm32"))]
-    pub fn num_threads(&mut self, num_threads: usize) -> Result<(), DiscreteLogError> {
+    pub fn num_threads(&mut self, num_threads: NonZeroUsize) -> Result<(), DiscreteLogError> {
         // number of threads must be a positive power-of-two integer
-        if num_threads == 0 || (num_threads & (num_threads - 1)) != 0 || num_threads > MAX_THREAD {
+        if !num_threads.is_power_of_two() || num_threads.get() > MAX_THREAD {
             return Err(DiscreteLogError::DiscreteLogThreads);
         }
 
-        self.num_threads = num_threads;
-        self.range_bound = (TWO16 as usize).checked_div(num_threads).unwrap();
-        self.step_point = Scalar::from(num_threads as u64) * G;
+        self.num_threads = Some(num_threads);
+        self.range_bound = (TWO16 as usize)
+            .checked_div(num_threads.get())
+            .and_then(|range_bound| range_bound.try_into().ok())
+            .unwrap(); // `num_threads` cannot exceed `TWO16`, so `range_bound` always positive
```

Comment:
nit: since this can only panic if `range_bound` is 0, and since "positive" could be misinterpreted (by me), how about we make that totally clear in the comment?
```suggestion
            .unwrap(); // `num_threads` cannot exceed `TWO16`, so `range_bound` always non-zero
```